File: model/response.go
package model

type Response struct {
	Code int `json:"code" example:"200"` //状态码，和http状态码剥离，以分离网络和业务
	Msg  any `json:"msg"`                //信息。可以是NULL，也可以是成功时返回的结构体，也可以是失败时的错误信息
}

File: frontend/submission-web/src/services/http/response.ts
import type { ApiResponse } from '@/types/api'
import { ApiError } from './errors'

export function unwrapResponse<T>(response: ApiResponse<T>) {
  if (response.code !== 200) {
    const detail = typeof response.msg === 'string' ? response.msg : '业务请求失败'
    throw new ApiError(detail, { code: response.code, detail: response.msg })
  }
  return response.msg
}
